File: equations/Lorenz.py
import numpy as np

def lorenz_equation(params=None):
    if params is None:
        params = {'gamma': 10, 'rho': 28, 'beta': 8. / 3}
        # params = {'gamma': 4, 'rho': 40, 'beta': 8}

    def lorenz(t, x, u):
        dx = np.zeros(3)
        dx[0] = params['gamma'] * (x[1] - x[0]) + u[0]*6
        dx[1] = x[0] * (params['rho'] - x[2]) - x[1] + 1.25*u[1]*x[1]
        dx[2] = x[0] * x[1] - params['beta'] * x[2] + 1.5*u[2]*x[2]
        return dx

        #dx1 = g*x2 - g*x1
        #dx2 = r*x1 - r*x3
        #dx3 = x1*x2 - b*x3

    return lorenz

File: src/lorenz/analytical_models/lorenz.m
function dxdt = lorenz(t,X,u,params)
%Input form (t, X, u, parameters)
    
    u = u(t);
    u1 = u(1);
    u2 = u(2);
    u3 = u(3);
    
    x = X(1);
    y = X(2);
    z = X(3);
    
%     dxdt = [params.sigma*(y-x) + u1,...
%         x*(params.rho - z) - y + 2*u2,...
%         x*y - params.beta*z + u1*x]';
%     
    dxdt = [params.sigma*(y-x) + 20*sin(u1),...
        x*(params.rho - z) - y + 3*u2,...
        x*y - params.beta*z + u1*x]';
    
